drivers/net/loopback.c

```diff
@@ -89,6 +89,19 @@ static int loopback_send(const struct device *dev, struct net_pkt *pkt)
 
 	ARG_UNUSED(dev);
 
+	if (IS_ENABLED(CONFIG_NET_LOOPBACK_SIMULATE_PACKET_DROP)) {
+		/* Drop packets based on the loopback_packet_drop_ratio
+		 * a ratio of 0.2 will drop one every 5 packets
+		 */
+		loopback_packet_drop_state += loopback_packet_drop_ratio;
+		if (loopback_packet_drop_state >= 1.0f) {
+			/* Administrate we dropped a packet */
+			loopback_packet_drop_state -= 1.0f;
+			loopback_packet_dropped_count++;
```

Comment:
Those variables are hidden under `#ifdef CONFIG_NET_LOOPBACK_SIMULATE_PACKET_DROP`, hence build failures in the CI. We need to either move them out, or stick to the `#ifdef CONFIG_NET_LOOPBACK_SIMULATE_PACKET_DROP` here as well.